library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt

```diff
@@ -1022,4 +1023,55 @@ class GroupTest {
         // Next syncAllGroups will not include the inactive group
         assertEquals(numGroups, 2u)
     }
+
+    @Test
+    fun testGroupDisappearingMessages() {
+        // Can enable message expiration on create
+        val boGroup = runBlocking {
+            boClient.conversations.newGroup(
+                listOf(alix.walletAddress),
+                messageDisappearingSettings = MessageDisappearingSettings(0L, 10L),
+            )
+        }
+        val messageId = runBlocking {
+            boGroup.send("howdy")
+        }
+        runBlocking { alixClient.conversations.syncAllConversations() }
+        var message = boClient.findMessage(messageId)
+        var alixMessage = alixClient.findMessage(messageId)
+        val alixGroup = alixClient.findGroup(boGroup.id)
+        assert(message != null)
+        assert(alixMessage != null)
+        assertEquals(runBlocking { boGroup.messages() }.size, 2)
+        assertEquals(runBlocking { alixGroup!!.messages() }.size, 1)
+        Thread.sleep(2000)
+        runBlocking {
+            boClient.conversations.syncAllConversations()
+            alixClient.conversations.syncAllConversations()
+        }
+        message = boClient.findMessage(messageId)
+        alixMessage = alixClient.findMessage(messageId)
+        assert(message == null)
+        assertEquals(runBlocking { boGroup.messages() }.size, 0)
+        assert(alixMessage == null)
+        assertEquals(runBlocking { alixGroup!!.messages() }.size, 0)
```

Comment:
@mchenani my messages aren't disappearing like I would expect is there something I need to do to trigger the worker to run?

Reply:
did you start the worker? and you need to wait for a second for the worker, the worker clean up the messages every second.